### ISSUE TITLE
Videos are Broken/Unplayable after exporting

### ISSUE THREAD

**Issue by jcgaming-official** [hardcore-sushi/DroidFS#245]
After I export the video. I tried to play it but it won't play. I have already tried different kinds of video player but nothing works. If the image is exported, the image is fine but the videos are broken. How to fix this?

**Comment by hardcore-sushi** [comment]
Where does the video comes from? Did you record a video from DroidFS or imported an already existing file?

Does it plays well in DroidFS? Can you send the video file?

**Comment by jcgaming-official** [comment]
> Where does the video comes from? Did you record a video from DroidFS or imported an already existing file?
> 
> Does it plays well in DroidFS? Can you send the video file?

The video is a screen record from my local device imported to DroidFS. Yes DroidFS works well to play videos but the exported is broken

**Comment by hardcore-sushi** [comment]
Check that the hash of the file after export is identical to that before import (you can use for example [this app](https://f-droid.org/en/packages/com.hobbyone.HashDroid)).

Can you play it with external video players by selecting "Open with external application" in DroidFS?

**Comment by jcgaming-official** [comment]
> Check that the hash of the file after export is identical to that before import (you can use for example [this app](https://f-droid.org/en/packages/com.hobbyone.HashDroid)).
> 
> Can you play it with external video players by selecting "Open with external application" in DroidFS?

How can I check the hash before import when I already deleted that file after import to DroidFs. Videos from DroidFs works well to external video player application

**Comment by jcgaming-official** [comment]
> > I already deleted that file after importing
> 
> Is the exported file the same size as the file in DroidFS? Can you reproduce the bug with another video? And maybe share it so I can inspect it?
> 
> The same code is used for exporting and opening files with external applications. This means that external apps should receive exactly the same bytes as what is written to disk when exporting a file. Can you show your logcat when exporting and opening the video from DroidFS?

It works now. The reason why it broken because when I decrypt and after locating where to save then it will go back do volume and I just exit without knowing there is a toast message saying "Export Successfully" so the exporting is interrupted.

 I have a suggestion:
Make a loading activity screen with percentage progress of exporting instead of using toast message. So the user will not exit and know how much time left until successfully export

**Comment by hardcore-sushi** [comment]
Yes that's already a todo feature ([TODO.md](https://github.com/hardcore-sushi/DroidFS/blob/master/TODO.md#ux)). At the moment it's shown as a notification.

**Comment by jcgaming-official** [comment]
Sir @hardcore-sushi help. My droidfs is broken. I'm using version 2.1.3. After I export many videos worth 3gb+ storage, my droidfs is now broken. I mean every time I opened the vault, the app kept lagging and kept popping up the "Droidfs is not responding (Close app or Wait), it's keep popping up if I click the wait button. I don't know how to fix it. I have 8+gb ram +5gb ram from additional virtual ram and my android version is 13

**Comment by jcgaming-official** [comment]
> Sir @hardcore-sushi help. My droidfs is broken. I'm using version 2.1.3. After I export many videos worth 3gb+ storage, my droidfs is now broken. I mean every time I opened the vault, the app kept lagging and kept popping up the "Droidfs is not responding (Close app or Wait), it's keep popping up if I click the wait button. I don't know how to fix it. I have 8+gb ram +5gb ram from additional virtual ram and my android version is 13

It's fixed now. Idk how. But how to fix this?
![Screenshot_20231031-122420.jpg](https://github.com/hardcore-sushi/DroidFS/assets/77482435/14c8add6-5620-4527-8b9a-de557541d057)

**Comment by jcgaming-official** [comment]
> Sir @hardcore-sushi help. My droidfs is broken. I'm using version 2.1.3. After I export many videos worth 3gb+ storage, my droidfs is now broken. I mean every time I opened the vault, the app kept lagging and kept popping up the "Droidfs is not responding (Close app or Wait), it's keep popping up if I click the wait button. I don't know how to fix it. I have 8+gb ram +5gb ram from additional virtual ram and my android version is 13

It's happening again.

**Comment by hardcore-sushi** [comment]
Does it keep not responding if you disable thumbnails and folder mapping? Is it a CryFS or a gocryptfs volume? How many files are present in the root directory?

**Comment by jcgaming-official** [comment]
> Does it keep not responding if you disable thumbnails and folder mapping? Is it a CryFS or a gocryptfs volume? How many files are present in the root directory?

It keeps not responding even I turned off the thumbnail and folder mapping. I used CryFS volume. There is only 2 folders in droidfs root directory and when I go to big folder it starts crashing

**Comment by hardcore-sushi** [comment]
> when I go to big folder it starts crashing

OK that's probably the problem. How many files are present in this big folder? Does the original CryFS program work fine when opening this folder on desktop?

If you create a gocryptfs volume and import the big folder in it, does the app crash when opening this folder on this gocryptfs volume?

**Comment by jcgaming-official** [comment]
Hi @hardcore-sushi. Time for feedback. Gocryptfs works really smooth on big volumes. I can even use thumbnail for big files. I don't know why cryfs is slow and not responding. I decided to move the files from cryfs to gocryptfs and I deleted already the cryfs. Thanks for help 🙂 I am waiting for new update

**Comment by hardcore-sushi** [comment]
gocryptfs is faster than CryFS: https://nuetzlich.net/gocryptfs/comparison#performance-on-linux However, DroidFS should not become unresponsive. I'll investigate the issue.
